src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/WorkflowGuardListenerPass.php

```diff
@@ -31,20 +31,18 @@ public function process(ContainerBuilder $container)
 
         $container->getParameterBag()->remove('workflow.has_guard_listeners');
 
-        if (!$container->has('security.token_storage')) {
-            throw new LogicException('The "security.token_storage" service is needed to be able to use the workflow guard listener.');
-        }
-
-        if (!$container->has('security.authorization_checker')) {
-            throw new LogicException('The "security.authorization_checker" service is needed to be able to use the workflow guard listener.');
-        }
-
-        if (!$container->has('security.authentication.trust_resolver')) {
-            throw new LogicException('The "security.authentication.trust_resolver" service is needed to be able to use the workflow guard listener.');
-        }
-
-        if (!$container->has('security.role_hierarchy')) {
-            throw new LogicException('The "security.role_hierarchy" service is needed to be able to use the workflow guard listener.');
+        $servicesNeeded = array(
+            'security.token_storage',
+            'security.authorization_checker',
+            'security.authentication.trust_resolver',
+            'security.role_hierarchy',
+            'validator',
```

Comment:
Are you suggesting to inject `null` instead of the `validator` service if this one is not available?
If yes, It's a good idea. I will update the PR.